spatialprofilingtoolbox/workflow/common/sparse_matrix_puller.py

```diff
@@ -118,10 +126,12 @@ def get_data_arrays(self):
         return self.data_arrays
 
     def _retrieve_data_arrays(self,
-            specimen: str | None=None,
-            study: str | None=None,
-            continuous_also: bool=False,
-        ) -> CompressedDataArrays:
+        specimen: str | None = None,
+        study: str | None = None,
+        continuous_also: bool = False,
+    ) -> CompressedDataArrays:
+        if specimen is not None:
+            study = StudyAccess(self.cursor).get_study_from_specimen(specimen)
```

Comment:
This should work to specialize to just the one study in case a specimen is provided.